### ISSUE TITLE
Migrate to vertx version 4

### ISSUE THREAD

**Issue by deedarb** [reactiverse/consul-cluster-manager#22]

**Comment by romalev** [comment]
yes indeed - this has to be done. @deedarb - feel free to contribute.

**Comment by deedarb** [comment]
i've seen hazelcast, infinispan implementation for vertx 4.
Planning to implement for consul, but not sure about **clusterHost** and **clusterPublicHost** in new ClusterManager SPI.
Should i implement them or no? 
Hazelcast starts "agent" with vertx instance, but consul do not, zookeeper is similar to consul and there is no implementations for those methods in zookeeper.

**Comment by deedarb** [comment]
@vietj could you help me with understanding clusterHost & clusterPublicHost, if i understand correctly 
`If the cluster manager has its own server for data/membership, this returns the host it is listening to. When users don't configure the eventbus cluster host, the value will serve as a default.
Returns:
null if the cluster manager does not start a server or the host couldn't be determined` 

vertx instance will not start consul agent so i should not override **clusterHost**?
`If the cluster manager has its own server for data/membership, this returns the host it advertises to other nodes. When users don't configure the eventbus cluster public host, the value will serve as a default.
Returns:
null if the cluster manager does not advertise a host that is different from the cluster host or the public host couldn't be determined`
this is not clear for me: how to decide if "cluster manager does not advertise a host that is different" ?

**Comment by pmlopes** [comment]
Tagging @tsegismont for his input here.

**Comment by tsegismont** [comment]
@deedarb some cloud providers have network configurations that make you bind server sockets to a hostname that is not the name used by others to reach you from the outside (for security reasons).

Vert.x needs to know which hostname (cluster host) to use when binding the eventbus server socket. It also needs to know which hostname (public host) it should advertise to other Vert.x nodes.

**Comment by robertonav20** [comment]
There is some update for this task?

**Comment by deedarb** [comment]
@robertonav20 I stopped migrating, planning to switch to quarkus + stork
